### ISSUE TITLE
Death penalties bug

### ISSUE THREAD

**Issue by Hexaae** [AFADoomer/Wolf3D-TC-3.0#14]
I've just learned from [manual of original game](http://www.wolfensteingoodies.com/wolfensteingoodies/docs/Wolfenstein_3D_-_Manual_-_PC.pdf) (see Lives paragraph) after you die you should restart current level with pistol only and 8 shots, and "you'll also lose the points you obtained on that level"...

**Comment by Hexaae** [comment]
Tested normal E1L2 and found the same issues: lives get reset to 3 when you die, and you won't lose weapons.

**Comment by Hexaae** [comment]
Tested original DOS game and indeed even at E1L2 if you die you'll restart with pistol + 8 bullets only.

**Comment by AFADoomer** [comment]
Ugh.  It works properly only on map 1.  *sigh*

Also: ESC cannot be disabled in GZDoom.  It's checked for in native code that runs before anything that I can do with ZScript would be called.

**Comment by Hexaae** [comment]
ESC:
mmmh, other possible ideas for workarounds: could you temporarily disable some menu options like New Game maybe?

**Comment by AFADoomer** [comment]
The incorrect life behavior part of restarting a new game while dead should be fixed now...

I just pushed more fixes for death resets in ef5e4bf8559078bb529f708f13dd6f122c76f4fc...

**Comment by Hexaae** [comment]
Mmmh, reached E1L2 with 4 lives, died and restarted with 4 lives :(

**Comment by AFADoomer** [comment]
Can you set the 'disableautosave' cvar to 1 and try again?

**Comment by Hexaae** [comment]
Done and started a new game from scratch. E1L2 with 4 lives, died and decreased correctly to 3, but pistol had 0 bullets.

**Comment by Hexaae** [comment]
A scripted autosave is required anyway when you reach a new level IMHO. I think it was normal behaviour of the original game.

**Comment by Hexaae** [comment]
There are three situations to check:
1. E1L1, save game and complete level. Die at E1L2 and test what happens. Then die again and see what happens.
2. E1L1 and complete level. Die at E1L2 ans see what happens.
3. E1L1. Die at least once. Complete level and see what happens loading new level.

**Comment by AFADoomer** [comment]
OK, I think this is as fixed as it will get as of ad603e64a1521204268edaf041d0a2c4f3d1a7e5.

If you never save manually, then the life counter works as expected - it will decrement when you die, and stay that way when you respawn, plus all of your weapons/ammo/other inventory will be reset (and the screen will fade in from red).

If you save manually, that save point will be used when you respawn, just as if you had reloaded manually - in this case, the save is loaded normally, so your life counter won't stay decremented and your weapons won't be stripped (and you won't get a fade-in from red).

If you save manually in one level, then finish the map, the next level will decrement everything normally (until/unless you save manually again)

This all depends on the 'disableautosaves' cvar being '0'.

**Comment by Hexaae** [comment]
> If you save manually, that save point will be used when you respawn, just as if you had reloaded manually
Can this be changed (or optional) to be closer to the original: you die, you respawn with penalties from the beginning of current level? 😇

**Comment by AFADoomer** [comment]
The behavior is hard-coded in GZDoom, unfortunately.  When you save a game, [that save game's name is saved and blindly/automatically used](https://github.com/ZDoom/gzdoom/blob/303e751a810318f7f4d0b4fbef36d73cf7b104de/src/g_game.cpp#L1725) as the reload map when you die, with no options for configuring the behavior.

**Comment by Hexaae** [comment]
Mmmh... I see...

Just experienced something strange BTW: saved game at E1L1, finished the level. Died at E1L2 and the game reloaded E1L1 savegame :( Can't be forced to restart/reset current level instead of following std GZDoom behavior?

**Comment by Hexaae** [comment]
Another case: completed E1L1 with 2 lives left. E1L2 started with 3 lives.

**Comment by AFADoomer** [comment]
> Just experienced something strange BTW...
Check your 'disableautosave' CVar value.  It needs to be set to zero.

> Another case: completed E1L1 with 2 lives left. E1L2 started with 3 lives.
Are you sure that you didn't get over 40000 points in bonus, which would give you a life?

**Comment by AFADoomer** [comment]
No, not without losing all points and saved level stats (so the end of game tally screen would be wrong)

**Comment by Hexaae** [comment]
Now here we go! MUCH better and closer to the original difficulty! 👍🏻

**Comment by Hexaae** [comment]
Mmmh, ok. So this can't be exactly the same of the original game. All right, after all everybody reloads the last quick-save indeed when they die...
P.S.
Can this feature (disable auto-load of last quicksave) be asked in the official GZDoom forum/github? Maybe they can implement it...

**Comment by AFADoomer** [comment]
Requested feature [here](https://forum.zdoom.org/viewtopic.php?t=76914).  We'll see if anything comes of it...

**Comment by nashmuhandes** [comment]
> Requested feature [here](https://forum.zdoom.org/viewtopic.php?t=76914). We'll see if anything comes of it...

Should be trivial to add to the engine, I'll see what I can do

**Comment by AFADoomer** [comment]
Resolved in 939720585e73823e3b0e99b2d130c5445bf0e56d.
